src/main/java/edu/harvard/iq/dataverse/util/SystemConfig.java

```diff
@@ -122,6 +122,11 @@ public class SystemConfig {
     public final static String DEFAULTCURATIONLABELSET = "DEFAULT";
     public final static String CURATIONLABELSDISABLED = "DISABLED";
     
+    // A secret used in signing URLs - individual urls are signed using this and the
+    // intended user's apiKey, creating an aggregate key that is unique to the user
+    // but not known to the user (as their apiKey is)
+    public final static String API_SIGNING_SECRET = "dataverse.api-signing-secret";
```

Comment:
OMG yes please! I'm happy to assist with this, too. I suppose we should create a new scope for API things, too, as I can use this for the various database settings regarding API functionality, too.

@qqmyers if you want me to create a commit or PR to your PR and add this for you, please ring a bell.

Reply:
I don't have a strong opinion about when/where this gets updated as long as it is backward compatible, which, if I understand it is because the base change is already in develop.

Reply:
I added a PR for this against the feature branch. See https://github.com/GlobalDataverseCommunityConsortium/dataverse/pull/31

I noted there is no documentation for the setting in the docs. Please let me know if I should change that, too, including a description how to deal with such a sensitive secret and MPCONFIG (making it easier).

Reply:
Sure that sounds useful - thanks!

Reply:
See https://github.com/GlobalDataverseCommunityConsortium/dataverse/pull/31/commits/fafa9b133db7dbe93733d24fa08ba162b9724bec